frontend/packages/ceph-storage-plugin/src/components/bucket-class/create-bc.tsx

```diff
@@ -43,32 +46,72 @@ const CreateBucketClass: React.FC<CreateBCProps> = ({ match }) => {
       .catch(() => setClusterServiceVersion(null));
   }, [appName, ns]);
 
-  const finalStep = () => {
-    dispatch({ type: 'setIsLoading', value: true });
-    const payload = {
-      apiVersion: apiVersionForModel(NooBaaBucketClassModel),
-      kind: NooBaaBucketClassModel.kind,
-      metadata: {
-        name: state.bucketClassName,
-        namespace: ns,
-      },
-      spec: {
-        placementPolicy: {
-          tiers: [
-            {
-              placement: state.tier1Policy,
-              backingStores: state.tier1BackingStore.map(getName),
-            },
-          ],
+  const getNamespaceStorePage = () => {
+    switch (state.namespacePolicyType) {
+      case NamespacePolicyType.SINGLE:
+        return <SingleNamespaceStorePage dispatch={dispatch} namespace={ns} />;
+      default:
+        return null;
+    }
+  };
+
+  const getPayload = () => {
+    let payload = null;
+    if (state.bucketClassType === BucketClassTypes.STANDARD) {
+      payload = {
+        apiVersion: apiVersionForModel(NooBaaBucketClassModel),
+        kind: NooBaaBucketClassModel.kind,
+        metadata: {
+          name: state.bucketClassName,
+          namespace: ns,
         },
-      },
-    };
-    if (state.tier2Policy) {
-      payload.spec.placementPolicy.tiers.push({
-        placement: state.tier2Policy,
-        backingStores: state.tier2BackingStore.map(getName),
-      });
+        spec: {
+          placementPolicy: {
+            tiers: [
+              {
+                placement: state.tier1Policy,
+                backingStores: state.tier1BackingStore.map(getName),
+              },
+            ],
+          },
+        },
+      };
+      if (state.tier2Policy) {
+        payload.spec.placementPolicy.tiers.push({
+          placement: state.tier2Policy,
+          backingStores: state.tier2BackingStore.map(getName),
+        });
+      }
+    } else {
+      switch (state.namespacePolicyType) {
+        case NamespacePolicyType.SINGLE:
+          payload = {
+            apiVersion: apiVersionForModel(NooBaaBucketClassModel),
+            kind: NooBaaBucketClassModel.kind,
+            metadata: {
+              name: state.bucketClassName,
+              namespace: ns,
+            },
+            spec: {
+              namespacePolicy: {
+                type: state.namespacePolicyType,
+                single: {
+                  resource: state.readNamespaceStore[0]?.metadata?.name,
+                },
+              },
+            },
+          };
+          break;
+        default:
+          return null;
+        // TODO Prepare payload for other policy types
+      }
     }
+    return payload;
+  };
+  const finalStep = () => {
+    dispatch({ type: 'setIsLoading', value: true });
+    const payload = getPayload();
```

Comment:
It would be nice, if you can make this function pure.
You can pass state in parameters.

Reply:
done